src/dfx/src/actors/pocketic.rs

```diff
@@ -238,41 +253,82 @@ fn pocketic_start_thread(
 ) -> DfxResult<std::thread::JoinHandle<()>> {
     let thread_handler = move || {
         loop {
-            // Start the process, then wait for the file.
-            let pocketic_path = config.pocketic_path.as_os_str();
+            let mut bind_address = config.pocketic_proxy_config.bind;
+            let (mut child, docker_container_id, last_start) =
+                if let Some(docker_image) = config.docker.as_ref() {
+                    // Run the container.
+                    // TODO: Check if docker command is available.
+                    let mut cmd = std::process::Command::new("docker");
+                    cmd.args(["run"]);
+                    bind_address = convert_to_docker_bind_address(bind_address);
+                    cmd.args([
+                        "-p",
+                        format!("{}:{}", bind_address.port(), bind_address.port()).as_str(),
+                        "-p",
+                        "8081:8081",
+                    ]);
+                    cmd.args(["-d", docker_image.as_str()]);
+                    cmd.stdout(std::process::Stdio::piped());
```

Comment:
I'd also expect the `-v` option to be used in order to mount the state to be persisted.